clients/src/test/java/org/apache/kafka/common/protocol/types/ProtocolSerializationTest.java

```diff
@@ -414,7 +416,7 @@ public void testReadWhenOptionalDataMissingAtTheEndIsNotTolerated() {
         oldFormat.writeTo(buffer);
         buffer.flip();
         SchemaException e = assertThrows(SchemaException.class, () -> newSchema.read(buffer));
-        e.getMessage().contains("Error reading field 'field2': java.nio.BufferUnderflowException");
+        assertThat(e.getMessage(), containsString("Error reading field 'field2': java.nio.BufferUnderflowException"));
```

Comment:
Is this error message from java standard library? If so, does it got different on different location (i.e different language)?

Reply:
The SchemaException is throw by KAFKA. 
it's message is  `fix string + field info   +  e.msg || e.className` 
the lastest string here is className   

but It's seems to me that only check the prefix is more better

Reply:
You are right. Checking ```Error reading field 'field2'``` should be enough for this test case.